src/cache_cf.cc

```diff
@@ -1120,9 +1120,9 @@ CheckTimeValue(const double value, const std::chrono::nanoseconds &unit)
     if (value < 0)
         throw TexcHere("time must have a positive value");
 
-    const auto maxNanoseconds = std::chrono::nanoseconds::max().count();
-    if (value > maxNanoseconds/static_cast<double>(unit.count())) {
-        const auto maxYears = maxNanoseconds/(HoursPerYear*3600*1000000000);
+    if (unit > std::chrono::nanoseconds::max()) {
```

Comment:
Spent some time digging the code. 
This bit is buggy (my fault) but the entire time parsing thing is bizarre.

I have serious doubts about us needing a nanoseconds time unit - especially considering that in parseTimeLine we issue a WARNING if it is used at all.
Then all time units are then cast to nono by CheckTimeValue, which is misnamed because it doesn't check but it's an integral part to the conversion process.
And then we cast back to the appropriate time unit.

Might as well deprecate nanoseconds as an unit, round to milliseconds instead, gaining 3 orders of magnitude of durations, and remove one pointless warning.

Fixing the most obvious problem in the meantime.

Reply:
> I have serious doubts about us needing a nanoseconds time unit

FYI: We need a nanoseconds time unit because of features like [paranoid_hit_validation ](https://github.com/squid-cache/squid/commit/b2aca62a222cafc8cffd4b1d09a5e1fd104d7612). I am pretty sure there are other existing and/or future features that would benefit from high-precision timers. The world is getting faster and some of the things that Squid does need to keep up with the current pace.

AFAICT, you have not disclosed the reason behind your serious nanosecond support doubts, but if you are concerned about the total supported duration, then you should not be:  The standard library [guarantees](https://en.cppreference.com/w/cpp/chrono/duration) a "range of at least ±292 years".

On the other hand, I totally agree that most duration-related parameters do not need to support nanoseconds and should not be converted to time units they do not support. IIRC, a recent 8da861a commit made significant progress in cleaning up related issues, but more painful work is needed in that direction.

Reply:
In this context, it makes sense to only have one internal representation of time. Let's keep it, just make it sane